src/d2_docker/utils.py

```diff
@@ -221,9 +223,12 @@ def run_docker_compose(
 
     env_pairs = [
         ("DHIS2_DATA_IMAGE", final_image_name),
-        ("DHIS2_CORE_PORT", str(port)) if port else None,
+        ("DHIS2_CORE_PORT", str(port) if port else None),
         ("DHIS2_CORE_IMAGE", core_image_name),
         ("LOAD_FROM_DATA", "yes" if load_from_data else "no"),
+        ("DB_DEBUG_PORT", str(db_debug_port) if db_debug_port else None),
+        ("CORE_DEBUG_PORT", str(core_debug_port) if core_debug_port else None),
```

Comment:
Not sure about how to do this:

- ``docker-compose`` does not allow conditional port forwarding and setting it to ''0'' or a negative value throws and error.

Possibilities:

- [uglier] We could have two ``docker-compose`` yaml files and add a flag ``--debug`` that it's more insecure.

- [complex] We could switch from ``docker-compose up`` to ``docker-compose run`` and handle the port forwarding in the Python code.

Reply:
Indeed, it's really hard to use docker-compose.yml when you have optional variables. A third option: create a docker-compose.yml on the fly with the env variables we are really going to use. That would also solve the empty dirs/files hack on the volumes sections.